pyquil/api/qvm.py

```diff
@@ -250,9 +280,13 @@ def expectation(self, prep_prog, operator_programs=None):
 
         :param Program prep_prog: Quil program for state preparation.
         :param list operator_programs: A list of PauliTerms. Default is Identity operator.
+        :param needs_compilation: If True, preprocesses the job with the compiler.
+        :param isa: If set, compiles to this target ISA.
         :returns: Expectation value of the operators.
         :rtype: float
         """
+        if needs_compilation:
```

Comment:
I think maybe we should just not expose this option for expectation if it doesn't work.

Reply:
The underlying issue here is the job payload: the compiler expects "uncompiled-quil" in and delivers on "compiled-quil", but the QVM looks in different slots for its incoming quil instructions depending on the program type. I'm more comfortable implicitly promising a fix to the QVM behavior by providing this API. @tarballs-are-good: is this reasonable?